Gene co-expression networks represent the interconnectedness between genes. The nodes are the genes, the edges are similarity/ co-expression relationships between them.
As we have seen in the last chapter [[Chapter 3 - Genetic expression]] we can now gather Gene expression data. We can see gene expression data as a score associated with each gene of each sample according to the gene expression value. We can call this the **Profile of expression values of a gene** and it can be represented with an heatmap where columns represents samples and colors represent gene expressions:![[Profile_of_expression_values_heatmap.png]]
In this heatmap the goal is to show the effect of a cancer disease. Fold change and log2ratio are used to visualize differential expression.
Gene networks and particularly co-expression networks provide the potential to identify hundreds of genes that are associated with complex human diseases and that could serve as points for therapeutic interventions. We can use this information for predicting functions of new genes and finding genes that play key roles in complex human diseases.  
## Similarity measure
We can also compute a **similarity measure** among gene expression profiles and visualize the results as a network. We can then identify genes with similar expression profiles this way.
Here's a list of the most typical similarity measures used:
- **Pearson's correlation**: measures the correspondence of two vectors (in this case genes). The pros are that is scalable and it can be efficiently computed for a large number of genes. However it's sensitive to outliers and it assumes that gene expression data follow a normal distribution;
- **Euclidean distance**: measures the geometric distance of two vectors. It's not appropriate when the absolute expression levels of functionally related genes are heavily different and if two genes are correlated randomly but have consistently low expression levels they might still appear close in the Euclidean space;
- **Mutual information**: measures how much the information of a gene reduces the uncertainty about the expression levels of another. It can detect non linear relationships, however some sophisticated non linear relationships might not even be relevant biologically. The distribution of data is also needed for the computation of mutual information while also needing a lot of samples for a good estimate;
- **Spearman's rank correlation**: is the Pearson's correlation calculated for the ranks of gene expression values in a gene expression vector. Is robust to outliers wrt standard Pearson correlation, but is less sensitive to expression values and with small number of samples it might detect false positives;
