DNA microarrays are basically orderly and miniaturized arrangements of fragments of DNA with known sequences on solid support. 
Each position on the support contains a fragment of DNA specific probe, complementary to the transcribed sequence. 
When the probe is placed near a complementary fragment (the test, marked with fluorochrome) it will tend to pair due to complementarity. This will activate the fluorochrome, and during the scan, the regions will have a fluorescent signal proportional to the amount of the correspondent transcript present in the biological sample used.
This tech can be used not only to measure genetic expression, but also as a measurement of abundance of genetic transcript, for characterization of a gene sequence, for characterization of alteration of the number of copies of a give gene or DNA sequence, for analysis of DNA to proteins interactions etc...
## DNA microarrays
Here the main steps:
1) Build the microarray (it's a solid holder where thousands of sequences of different genes, called **probes** are placed in known locations)
2) Extract RNA from the cells that you want to examine (this RNA is called test), reverse transcript it into cDNA if needed and amplify and mark it
3) Hybridize the test to the microarray
4) Evaluate the gene activity

This method is game-changer, since it allowed to change the analysis from one or some genes to thousand of genes at once.
## cDNA microarrays
1) full sections of ESTs (expressed sequence tags, which are short subsequences of a transcribed cDNA sequence) are copied. Each EST should be specific. 
2) Then two mRNA samples are prepared and retro-transcribed into cDNA (and marked fluorescent with different colors)
3) gene transcripts expressed in sample, prepared and marked are hybridized with their complementary sequence on the microarray
4) The fluorescent marker is measured in each spot to get which gene are expressed in each of the two samples.
Each slide can have up to 15k elements, but usually its 5k. The slides are printed by robots.
Due to the setup they are quite expensive (it takes 2 days to produce around 100 microarrays).
Usually the image of the microarray with fluorescent markers is taken and processed by a computer:
- each probe is localized in the array image
- each pixel is classified to belong to the spot it corresponds to
- The intensity of color is extracted to obtain a numeric value that represents the expression
- The background intensity is extracted and removed from the intensity values
- The quality of the experiment is verified using specific parameters
## Oligonucleotide microarrays
In this method oligonucleotides replace ESTs. oligonucleotides are sequences of bases of length 20-80.
The composition of each set of nucleotide is the following: for each probe with sequence of **Perfect matches (PM)** there's (on the chip) another probe with a sequence of **Mismatch (MM)**.
Each gene is represented by a set of 10-20 oligonucleotides corresponding to some positions of the represented gene, each with PM and MM.
The oligonucleotides are synthetized on a silicon chip using lithography by specialized companies.
The conversion from image to data is the same used for cDNA microarrays. To note that mRNA is used as test instead of cDNA.

## cDNA vs oligonucleotides
cDNA:
- do not need the complete genome to be applied to an organism
- cheaper (but still expensive)
- more flexible and rely on hybridization between many bases (instead of few ones)Ã¹
- probes are cDNA
oligonucleotides:
- can contain higher amount of genes (even predicted ones) that are not yet inserted in a cDNA library
- Only for sequenced organisms
- can be used by who cannot build a slide
- have less variability between different chips
- probes are short sequences designed to match parts of the sequences of known or predicted open reading frames of the gene
- for the above reason is easier to compare data generated by different research groups
## Microarray data and annotations
To simplify DNA microarray experiments and data annotation the MGED (microarray and gene expression data) was created. They created a **glossary** called MIAME (minimum information about a microarray experiment) and some other "derivates". 
The principles of MIAME are:
- Information acquired should be enough to interpret the results and replicate the experiments;
- information structure should allow queries and automatic analysis on data;

## Acquisition and pre-processing
After acquiring the fluorescent intensity there's a need to process the acquired data through:
- image analysis: identify probe positions and relate them to every gene (called **gridding**), then distinguish pixels related to background and foreground (segmentation) then extract intensity values from the image.
- data normalization: remove systematic errors and normalize the data based on a set of genes (called **house keeping genes**) which have invariant expression in different experimental conditions
- data transformation: apply outliers detection, logarithmic transformation and management of missing values 
- additional analysis: through data mining there can be extracted additional information. 
